src/app/email-validator/email.validator.ts

```diff
@@ -1,17 +1,16 @@
-import {Control} from '@angular/common';
-interface IValidationResult {
-  [key: string]: boolean;
-}
+import {FormControl} from "@angular/forms";
 
 export class EmailValidator {
-  static validEmail(control: Control): IValidationResult {
+  static validEmail(control: FormControl) {
     let value = control.value;
     /* tslint:disable */
-    let re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    let EMAIL_REGEXP = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
     /* tslint:enable */
-    if (value.length <= 5 || !re.test(control.value)) {
+    if (value.length <= 5 || !EMAIL_REGEXP.test(control.value)) {
       return {
-        invalidEmail: true
+        validEmail: {
+          invalidEmail: true
```

Comment:
is this a requirement? I don't see the need. Looks like a grouping that doesn't make sense to me.